README.md
<p>
<img title="Lazerpay" src= "https://res.cloudinary.com/njokuscript/image/upload/v1646279538/lazerpay_logo_no-bg_trkkye.png" width="300px"/>
</p>

#### The Lazerpay v1 Python SDK

## Installation

Use the package manager [pip](https://pip.pypa.io/en/stable/) to install LazerPayFinance.

```bash
pip install LazerPayFinance
```
If the approach is not working try this
```bash
py -m pip install LazerPayFinance
```
```python
from LazerPayFinance import LazerPayFinance
lazerpay = LazerPayFinance(LAZER_SECRET_KEY,LAZER_PUBLIC_KEY,True,'Binance Smart Chain')
```
The function above takes four parameters
##### LAZER_PUBLIC_KEY
##### LAZER_SECRET_KEY
##### Partial Payment (True,False)
##### Current Supported Network

Use TEST API keys for testing, and LIVE API keys for production

## Lazerpay Methods exposed by the sdk

**1**. **Payment**

- Initialize Payment
- Confirm Payment

**2**. **Payout**

- Crypto Payout
- Bank Payout ~ This is coming to V2

**3**. **Swap**

- Crypto swap
- Get Crypto Swap Amount Out

**4**. **Payment Links**

- Create payment links
- Get all payment links
- Get a single payment link
- Update a payment Link

**5**. **Misc**

- Get all accepted coins
- Get wallet balance

**6**. **Wallet**

- Get random wallet address
- Get random reference

## Payment

#### `Initialize Payment`
This describes to allow your customers to initiate a crypto payment transfer.
```python
import LazerPayFinance
lazerpay = LazerPayFinance(LAZER_PUBLIC_KEY, LAZER_SECRET_KEY,True,'Binance Smart Chain')

initializePayment = {
            'reference':82927118-282816-1,
            'customer_name':'Chimdindu',
            'customer_email':'chimdi4332@gmail.com',
            'coin':'USDC',
            'currency':'NGN',
            'amount':10,
            'accept_partial_payment':True
        }
initializePayment = lazerpay.initializePayment(initializePayment )
print("initializePayment =>")
print(initializePayment)
```
#### `Confirm Payment`

This describes to allow you confirm your customers transaction after payment has been made.
```python
verifyPayment = lazerpay.verifyPayment(ref)
print("verifyPayment =>")
print(verifyPayment)
```
## Transfer

#### `Crypto Payout`

This describes to allow you withdraw the crypto in their lazerpay balance to an external address
```python
p_4 = {
            'reference':'92992-82u2892-0800',
            'amount':10.00,
            'recipient':'0xD275Da57B35089a51A3E4659AcDF13311Ebd6183',#hex value
            'coin':'USDT',
            'metadata':{
                'type':'Crypto transfer'
            },
            'blockchain':'Binance Smart Chain'
        }
transferCrypto = lazerpay.transferCrypto(p_4)
print("transferCrypto =>")
print(transferCrypto)
```
## Swap

#### `Crypto swap`

This describes to allow you swap swap between two stable coins 
```python
p_5 = {
            'reference':'just a string',
            'amount':10.00,
            'fromCoin':'USDC',
            'toCoin':'USDT',
            'metadata':{
                'type':'Crypto Swap'
            },
            'blockchain':'Binance Smart Chain'
        }
swapCrypto = lazerpay.swapCrypto(p_5)
print("swapCrypto = >")
print(swapCrypto)
```
#### `Get Crypto Swap Amount Out`

This describes the amount you will receive on swap even before initiating the swap  
```python
p_6 = {
            'reference':'just a string',
            'amount':10.00,
            'fromCoin':'USDC',
            'toCoin':'USDT',
            'metadata':{
                'type':'Crypto Swap'
            },
            'blockchain':'Binance Smart Chain'
        }
swapCharge = lazerpay.swapCharge(p_6)
print("swapCharge =>")
print(swapCharge)
```
## Payment Links

#### `Create a payment link`

This describes to allow you create a Payment link programatically
```python
p_2 = {
        'title':'Hello this a new link',
        'description':'Description of payment',
        'amount':450,
        'type:':'standard',
        'logo':'https://www.thisisjustalogo.com',
        'currency':'NGN',
        'redirect_url':'https://hesichimdi.xyz'
        }
createLink =  lazerpay.createLink(p_2)
print("createLink =>")
print(createLink)
```
#### `Update a payment link`

This describes disabling or enabling a payment link by updating it
```python
p_3 = {
            'status':'active',
        }
updateLink = lazerpay.updateLink(ref,p_3)
print("updateLink =>")
print(updateLink)
```
#### `Get all payment links`

This describes to allow you get all Payment links created
```python
previousLink = lazerpay.previousLink()
print("previousLink =>")
print(previousLink)
```
#### `Get a single payment link`

This describes to allow you get a Payment link by it's identifier
```python
FetchSingleLink = lazerpay.FetchSingleLink(ref)#reference string
print("FetchSingleLink =>")
print(FetchSingleLink)
```
## Misc

#### `Get Accepted Coins`

This gets the list of accepted cryptocurrencies on Lazerpay
```python
getCoins = lazerpay.getCoins()
print("getCoins =>")
print(getCoins)
```
#### `Get Wallet Balance`

Get get wallet balance by specifying the coin
```python
walletbalance = lazerpay.walletBalance('USDT')
print("wallet_balance =>")
print(walletbalance)
```
#### `Get Wallet Balance`

Get get wallet balance by specifying the Address
```python
getBalance = lazerpay.getBalance(ADRESS,"to") #toWei. Use 'from' to convert from wei to ether
print("getBalance =>")
print(getBalance)
```
#### `Get Random Address`

This method get random address on the current blockChain specified
```python
createRandomAccount = lazerpay.createRandomAccount()
print('createRandomAccount=>')
print(createRandomAccount)
```
#### `Get Rate`

This method get the rate of a particular crypto to a specified currency e.g NGN,USD .e.t.c
```python
getRate = lazerpay.getRate('USDT','NGN')
print("getRate =>")
print(getRate)
```
#### `Get Accepted Fiats`

This gets the list of accepted fiat on Lazerpay

```python
getFiats = lazerpay.getFiats()
print("getFiats =>")
print(getFiats)
```
#### `Creating random string`
This function creates a random string and takes a single parameter as argument.
```python
ref = lazerpay.createReferenceString(20)#Length of string
print("ref => " + str(ref))
print("__DONE__")
```
## Contributing
Pull requests are welcome. For major changes, please open an issue first to discuss what you would like to change.

Please make sure to update tests as appropriate.
## `Example`
To view a full example on how to use this sdk, click
[here](https://github.com/Coderdivine/Lazerpay-python-sdk/blob/main/Example.py)
## `License`
[MIT](https://choosealicense.com/licenses/mit/)


CHANGELOG.txt
Change Log
==========

0.0.1 (19/04/2022)
------------------

- First Release

init.py
import requests
import numpy as np
import pandas as pd
import string    
import random # define the random module
#0x337610d27c682E347C9cD60BD4b3b107C9d34dDd
#0xD275Da57B35089a51A3E4659AcDF13311Ebd6183
class LazerPayFinance():
    def __init__(self,secretKey,publicKey,Partial,chain):
        self.chain = chain
        self.secretKey = secretKey
        self.publicKey = publicKey
        self.publicKey = publicKey
        self.Partial = Partial
    def walletBalance(self,coin):
        #The coin balance you want to get, eg USDT, DAI, BUSD, USDC
        url = 'https://api.lazerpay.engineering/api/v1/wallet/balance/?coins=USDC'
        #requesting = requests.Session()
        headers = {
            'Content-type':'application/json',
            'Authorization':self.secretKey
            }
        #requesting.headers.update({'Authorization':self.secretKey})
        response = requests.get(url,headers=headers)
        return response.json()
    def Authentication(self,method,url,params,type):
        requesting = requests.Session()
        #requesting.auth = ('user', 'pass')
        if type == "public":
           requesting.headers.update({'x-api-key':self.publicKey})
        else:
           requesting.headers.update({'x-api-key':self.secretKey})
        if method == "POST":
            r = requesting.post(url,params)
            return r.json()
        else:
            r = requesting.get(url)
            return r.json()
    def initializePayment(self,params):
        url = 'https://api.lazerpay.engineering/api/v1/transaction/initialize'
        # params = {
        #     'reference':'',
        #     'customer_name':'',
        #     'customer_email':'',
        #     'coin':'',
        #     'currency':'',
        #     'amount':10,
        #     'accept_partial_payment':self.Partial
        # }
        #requesting = requests.Session()
        headers = {
            'Content-type':'application/json',
            'x-api-key':self.publicKey
        }
        #requesting.headers.update({'x-api-key':self.publicKey})
        response = requests.post(url,json=params,headers=headers)
        return response.json()
    def getFiats(self):
        #The amount field should be in your fiat currency. 
        # The currencies we support are USD, AED, GBP, EUR and NGN
        response = {
            0:'USD',
            1:'AED',
            2:'GBP',
            3:'EUR',
            4:'NGN'
        }
        return response
    def getCoins(self):
        url = 'https://api.lazerpay.engineering/api/v1/coins'
        headers = {
            'Content-type':'application/json',
            'x-api-key':self.publicKey
        }
        # requesting = requests.Session()
        # requesting.headers.update({'x-api-key':self.publicKey})
        response = requests.get(url,headers=headers)
        return response.json()
    def createLink(self,params):
        url = 'https://api.lazerpay.engineering/api/v1/payment-links'
        # params = {
        #     'title':'hello',
        #     'description':'this is hello',
        #     'type':'standard',
        #     'currency':'USD',
        #     'redirect_url':'https://docs.lazerpay.finance'
        # }
        # requesting = requests.Session()
        # requesting.headers.update({'Authorization':self.secretKey})
        headers = {
            'Content-type':'application/json',
            'authorization':self.secretKey
        }
        response = requests.post(url,json=params,headers=headers)
        return response.json()
    def verifyPayment(self,reference_address):
        url = 'https://api.lazerpay.engineering/api/v1/transaction/verify/'+reference_address
        # requesting = requests.Session()
        # requesting.headers.update({'x-api-key':self.publicKey})
        print(url)
        headers = {
            'Content-type':'application/json',
            'x-api-key':self.publicKey
        }
        response = requests.get(url,headers=headers)
        return response.json()
    def createReferenceString(self,length):  
        # call random.choices() string module to find the string in Uppercase + numeric data.  
        ran = ''.join(random.choices(string.ascii_uppercase + string.digits, k = length))    
        return str(ran)
    def previousLink(self):
        url = 'https://api.lazerpay.engineering/api/v1/payment-links'
        # requesting = requests.Session()
        # requesting.headers.update({'Authorization':self.secretKey})
        headers = {
            'Content-type':'application/json',
            'Authorization':self.secretKey
        }
        response = requests.get(url,headers=headers)
        return response.json()
    def FetchSingleLink(self,reference):
        url = 'https://api.lazerpay.engineering/api/v1/payment-links/'+reference
        # requesting = requests.Session()
        # requesting.headers.update({'Authorization':self.secretKey})
        headers = {
            'Content-type':'application/json',
            'Authorization':self.secretKey
        }
        response = requests.get(url,headers=headers)
        return response.json()
    def updateLink(self,reference,params):
        url = 'https://api.lazerpay.engineering/api/v1/payment-links/'+reference
        # requesting = requests.Session()
        # requesting.headers.update({'Authorization':self.secretKey})
        headers = {
            'Content-type':'application/json',
            'Authorization':self.secretKey
        }
        # params = {
        #     'status':'active',
        # }
        response = requests.put(url,json=params,headers=headers)
        return response.json()
    def transferCrypto(self,params):
        url = 'https://api.lazerpay.engineering/api/v1/transfer'
        # params = {
        #     'reference':'92992-82u2892-000',
        #     'amount':'10',
        #     'recipient':'0xf000...',
        #     'coin':'USDT',
        #     'metadata':{
        #         'type':'Crypto transfer'
        #     },
        #     'blockchain':'Binance Smart Chain'
        # }
        # requesting = requests.Session()
        # requesting.headers.update({'Authorization':self.secretKey})
        headers = {
            'Content-type':'application/json',
            'Authorization':self.secretKey
        }
        response = requests.post(url,json=params,headers=headers)
        return response.json()
    def swapCrypto(self,params):
        url = 'https://api.lazerpay.engineering/api/v1/swap/crypto'
        # params = {
        #     'reference':'just a string',
        #     'amount':'10',
        #     'fromCoin':'USDC',
        #     'toCoin':'USDT',
        #     'metadata':{
        #         'type':'Crypto Swap'
        #     },
        #     'blockchain':'Binance Smart Chain'
        # }
        # requesting = requests.Session()
        # requesting.headers.update({'Authorization':self.secretKey})
        headers = {
            'Content-type':'application/json',
            'Authorization':self.secretKey
        }
        response = requests.post(url,json=params,headers=headers)
        return response.json()
    def swapCharge(self,params):
        url = 'https://api.lazerpay.engineering/api/v1/swap/crypto/amount-out'
        # params = {
        #     'amount':'10',
        #     'fromCoin':'DAI',
        #     'toCoin':'USDT',
        #     'blockchain':'Binance Smart Chain'
        # }
        # requesting = requests.Session()
        # requesting.headers.update({'Authorization':self.secretKey})
        headers = {
            'Content-type':'application/json',
            'Authorization':self.secretKey
        }
        response = requests.post(url,json=params,headers=headers)
        return response.json()
    def getRate(self):
        url = 'https://api.lazerpay.engineering/api/v1/rate'
        requesting = requests.Session()
        requesting.headers.update({'Authentication':self.publicKey})
        # headers = {
        #     'Content-type':'application/json',
        #     'Authorization':self.publicKey
        # }
        response = requesting.get(url)
        return response.json()
print('___tesing___')
lazerpay = LazerPayFinance('sk_test_Fh4xj28OXmTO3Ou08zu9xl9h4nslTysGugQI0l3s6J2bdT5T1I','pk_test_0bE5tKpLON8OGVdmjqAxL0woAwCEth65Lkg9hdxr9jvvCiiKv4',True,'Binance Smart Chain')
ref = lazerpay.createReferenceString(20)
print("ref => " + str(ref))
walletbalance = lazerpay.walletBalance('USDT')
print("wallet_balance =>")
print(walletbalance)
#initializePayment
p_1 = {
            'reference':ref,
            'customer_name':'Chimdi',
            'customer_email':'chimdi4332@gmail.com',
            'coin':'USDC',
            'currency':'NGN',
            'amount':10,
            'accept_partial_payment':True
        }
#initializePayment = lazerpay.initializePayment(p_1)
#print("initializePayment =>")
#print(initializePayment)
#getFiats = lazerpay.getFiats()
#print("getFiats =>")
#print(getFiats)
#getCoins = lazerpay.getCoins()
#print("getCoins =>")
#print(getCoins)
p_2 = {
            'title':'hello',
            'description':'this is hello',
            'type':'standard',
            'currency':'USD',
            'redirect_url':'https://docs.lazerpay.finance'
        }
createLink = lazerpay.createLink(p_2)
print("createLink =>")
print(createLink)
#verifyPayment = lazerpay.verifyPayment(ref)
#print("verifyPayment =>")
#print(verifyPayment)
previousLink = lazerpay.previousLink()
print("previousLink =>")
print(previousLink)
FetchSingleLink = lazerpay.FetchSingleLink(ref)
print("FetchSingleLink =>")
print(FetchSingleLink)
p_3 = {
            'status':'active',
        }
updateLink = lazerpay.updateLink(ref,p_3)
print("updateLink =>")
print(updateLink)
p_4 = {
            'reference':'92992-82u2892-000',
            'amount':'10',
            'recipient':'0xf000...',
            'coin':'USDT',
            'metadata':{
                'type':'Crypto transfer'
            },
            'blockchain':'Binance Smart Chain'
        }
transferCrypto = lazerpay.transferCrypto(p_4)
print("transferCrypto =>")
print(transferCrypto)
p_5 = {
            'reference':'just a string',
            'amount':'10',
            'fromCoin':'USDC',
            'toCoin':'USDT',
            'metadata':{
                'type':'Crypto Swap'
            },
            'blockchain':'Binance Smart Chain'
        }
swapCrypto = lazerpay.swapCrypto(p_5)
print("swapCrypto = >")
print(swapCrypto)
p_6 = {
            'reference':'just a string',
            'amount':'10',
            'fromCoin':'USDC',
            'toCoin':'USDT',
            'metadata':{
                'type':'Crypto Swap'
            },
            'blockchain':'Binance Smart Chain'
        }
swapCharge = lazerpay.swapCharge(p_6)
print("swapCharge =>")
print(swapCharge)
getRate = lazerpay.getRate()
print("getRate =>")
print(getRate)
print("__DONE__")




__init__.py
import requests
import string    
import random # define the random module
#0x337610d27c682E347C9cD60BD4b3b107C9d34dDd
#0xD275Da57B35089a51A3E4659AcDF13311Ebd6183
from Web3Package import WebProvider
WebProviders = WebProvider()
class LazerPayFinance():
    def __init__(self,secretKey,publicKey,Partial,chain):
        self.chain = chain
        self.secretKey = secretKey
        self.publicKey = publicKey
        self.publicKey = publicKey
        self.Partial = Partial
    def walletBalance(self,coin):
        #The coin balance you want to get, eg USDT, DAI, BUSD, USDC
        url = 'https://api.lazerpay.engineering/api/v1/wallet/balance/?coin=USDT'
        #requesting = requests.Session()
        headers = {
            'Content-type':'application/json',
            'Authorization':'Bearer '+self.secretKey
            }
        #requesting.headers.update({'Authorization':self.secretKey})
        response = requests.get(url,headers=headers)
        return response.json()
    def Authentication(self,method,url,params,type):
        requesting = requests.Session()
        #requesting.auth = ('user', 'pass')
        if type == "public":
           requesting.headers.update({'x-api-key':self.publicKey})
        else:
           requesting.headers.update({'x-api-key':'Bearer '+self.secretKey})
        if method == "POST":
            r = requesting.post(url,params)
           # return r.json()
        else:
            r = requesting.get(url)
            return r.json()
    def initializePayment(self,params):
        url = 'https://api.lazerpay.engineering/api/v1/transaction/initialize'
        # params = {
        #     'reference':'',
        #     'customer_name':'',
        #     'customer_email':'',
        #     'coin':'',
        #     'currency':'',
        #     'amount':10,
        #     'accept_partial_payment':self.Partial
        # }
        #requesting = requests.Session()
        headers = {
            'Content-type':'application/json',
            'x-api-key':self.publicKey
        }
        #requesting.headers.update({'x-api-key':self.publicKey})
        response = requests.post(url,json=params,headers=headers)
        return response.json()
    def getFiats(self):
        #The amount field should be in your fiat currency. 
        # The currencies we support are USD, AED, GBP, EUR and NGN
        response = {
            0:'USD',
            1:'AED',
            2:'GBP',
            3:'EUR',
            4:'NGN'
        }
        return response
    def getCoins(self):
        url = 'https://api.lazerpay.engineering/api/v1/coins'
        headers = {
            'Content-type':'application/json',
            'x-api-key':self.publicKey
        }
        # requesting = requests.Session()
        # requesting.headers.update({'x-api-key':self.publicKey})
        response = requests.get(url,headers=headers)
        return response.json()
    def createLink(self,params):
        url = 'https://api.lazerpay.engineering/api/v1/payment-links'
        # params = {
        # 'title':'Hello this a new link',
        # 'description':'Description of payment',
        # 'amount':450,
        # 'type:':'standard',
        # 'logo':'this is just a logo link',
        # 'currency':'NGN',
        # 'redirect_url':'https:hesichimdi.xyz'
        # }
        # requesting = requests.Session()
        # requesting.headers.update({'Authorization':self.secretKey})
        headers = {
            'Content-type':'application/json',
            'authorization':'Bearer '+self.secretKey
        }
        response = requests.post(url,json=params,headers=headers)
        return response.json()
    def verifyPayment(self,reference_address):
        url = 'https://api.lazerpay.engineering/api/v1/transaction/verify/'+reference_address
        # requesting = requests.Session()
        # requesting.headers.update({'x-api-key':self.publicKey})
        print(url)
        headers = {
            'Content-type':'application/json',
            'x-api-key':self.publicKey
        }
        response = requests.get(url,headers=headers)
        return response.json()
    def createReferenceString(self,length):  
        # call random.choices() string module to find the string in Uppercase + numeric data.  
        ran = ''.join(random.choices(string.ascii_uppercase + string.digits, k = length))    
        return str(ran)
    def previousLink(self):
        url = 'https://api.lazerpay.engineering/api/v1/payment-links'
        # requesting = requests.Session()
        # requesting.headers.update({'Authorization':self.secretKey})
        headers = {
            'Content-type':'application/json',
            'Authorization':'Bearer '+self.secretKey
        }
        response = requests.get(url,headers=headers)
        return response.json()
    def FetchSingleLink(self,reference):
        url = 'https://api.lazerpay.engineering/api/v1/payment-links/'+reference
        # requesting = requests.Session()
        # requesting.headers.update({'Authorization':self.secretKey})
        headers = {
            'Content-type':'application/json',
            'Authorization':'Bearer '+self.secretKey
        }
        response = requests.get(url,headers=headers)
        return response.json()
    def updateLink(self,reference,params):
        url = 'https://api.lazerpay.engineering/api/v1/payment-links/'+reference
        # requesting = requests.Session()
        # requesting.headers.update({'Authorization':self.secretKey})
        headers = {
            'Content-type':'application/json',
            'Authorization':'Bearer '+self.secretKey
        }
        # params = {
        #     'status':'active',
        # }
        response = requests.put(url,json=params,headers=headers)
        return response.json()
    def transferCrypto(self,params):
        url = 'https://api.lazerpay.engineering/api/v1/transfer'
        # params = {
        #     'reference':'92992-82u2892-000',
        #     'amount':'10',
        #     'recipient':'0xf000...',
        #     'coin':'USDT',
        #     'metadata':{
        #         'type':'Crypto transfer'
        #     },
        #     'blockchain':'Binance Smart Chain'
        # }
        # requesting = requests.Session()
        # requesting.headers.update({'Authorization':self.secretKey})
        headers = {
            'Content-type':'application/json',
            'Authorization':'Bearer '+self.secretKey
        }
        response = requests.post(url,json=params,headers=headers)
        return response.json()
    def swapCrypto(self,params):
        url = 'https://api.lazerpay.engineering/api/v1/swap/crypto'
        # params = {
        #     'reference':'just a string',
        #     'amount':'10',
        #     'fromCoin':'USDC',
        #     'toCoin':'USDT',
        #     'metadata':{
        #         'type':'Crypto Swap'
        #     },
        #     'blockchain':'Binance Smart Chain'
        # }
        headers = {
            'Content-type':'application/json',
            'Authorization':'Bearer '+self.secretKey
        }
        response = requests.post(url,json=params,headers=headers)
        return response.json()
    def swapCharge(self,params):
        url = 'https://api.lazerpay.engineering/api/v1/swap/crypto/amount-out'
        # params = {
        #     'amount':'10',
        #     'fromCoin':'DAI',
        #     'toCoin':'USDT',
        #     'blockchain':'Binance Smart Chain'
        # }
        # requesting = requests.Session()
        # requesting.headers.update({'Authorization':self.secretKey})
        headers = {
            #'Content-type':'application/json',
            'Authorization':'Bearer '+self.secretKey
        }
        response = requests.post(url,json=params,headers=headers)
        return response.json()
    def getRate(self,coin,currency):
        url = "https://api.lazerpay.engineering/api/v1/rate?coin={}&currency={}".format(coin,currency)
        # requesting = requests.Session()
        # requesting.headers.update({'x-api-key':self.publicKey})
        headers = {
            'Content-type':'application/json',
            'x-api-key':self.publicKey
        }
        response = requests.get(url,headers=headers)
        return response.json()
    def createRandomAccount(self):
        acc = WebProviders.getAccounts()
        return acc
    def getBalance(self,address,type):
        bal = WebProviders.getBalance(address,type)
        return bal
    def createAccount(self):
        acc = WebProviders.createAccount()
        return acc
print('___tesing___')
# py -m pip install setuptools twine
# py setup.py sdist
# twine upload --repository-url https://upload.pypi.org/legacy/ dist/*
print('__DONE___')



Web3Package.py
from os import access
from web3 import Web3
web3 = Web3(Web3.HTTPProvider("https://bsc-dataseed4.binance.org"))
print(web3.isConnected())
#no params
class WebProvider:
    def __init__(self):
        pass
    def BlockNumber(self):
       block =  web3.eth.blockNumber()
       return block
    def getBalance(self,address,type): 
        balance  = web3.eth.getBalance(address)
        if type == 'from':
            return web3.fromWei(balance, 'ether')
        elif type == 'to':
            return web3.toWei(balance,'ether')
        else:
            return balance
    def LatestBlock(self):
        block = web3.eth.get_block('latest')
        return block
    def getGasPrice(self):
        price = web3.eth.gas_price
        return price
    def getAccounts(self):
        acc = web3.eth.accounts
        return acc
    def getChainId(self):
        chain =  web3.eth.chain_id
        return chain
    def createAccount():
        acc = web3.eth.accounts.create()
        return acc


Example.py
from LazerPayFinance import LazerPayFinance
lazerpay = LazerPayFinance('sk_test_Fh4xj28OXmTO3Ou08zu9xl9h4nslTysGugQI0l3s6J2bdT5T1I','pk_test_0bE5tKpLON8OGVdmjqAxL0woAwCEth65Lkg9hdxr9jvvCiiKv4',True,'Binance Smart Chain')
ref = lazerpay.createReferenceString(20)
print("ref => " + str(ref))
walletbalance = lazerpay.walletBalance('USDT')
print("wallet_balance =>")
print(walletbalance)
#initializePayment
p_1 = {
            'reference':ref,
            'customer_name':'Chimdi',
            'customer_email':'chimdi4332@gmail.com',
            'coin':'USDC',
            'currency':'NGN',
            'amount':10,
            'accept_partial_payment':True
        }
#initializePayment = lazerpay.initializePayment(p_1)
#print("initializePayment =>")
#print(initializePayment)
#getFiats = lazerpay.getFiats()
#print("getFiats =>")
#print(getFiats)
#getCoins = lazerpay.getCoins()
#print("getCoins =>")
#print(getCoins)
p_2 = {
        'title':'Hello this a new link',
        'description':'Description of payment',
        'amount':450,
        'type:':'standard',
        'logo':'https://www.thisisjustalogo.com',
        'currency':'NGN',
        'redirect_url':'https://hesichimdi.xyz'
        }
createLink =  lazerpay.createLink(p_2)
print("createLink =>")
# print(createLink)
ref = createLink['data']['reference']
#verifyPayment = lazerpay.verifyPayment(ref)
#print("verifyPayment =>")
#print(verifyPayment)
previousLink = lazerpay.previousLink()
print("previousLink =>")
print(previousLink)
FetchSingleLink = lazerpay.FetchSingleLink(ref)
print("FetchSingleLink =>")
print(FetchSingleLink)
p_3 = {
            'status':'active',
        }
updateLink = lazerpay.updateLink(ref,p_3)
print("updateLink =>")
print(updateLink)
#done
p_4 = {
            'reference':'92992-82u2892-0800',
            'amount':10.00,
            'recipient':'0xD275Da57B35089a51A3E4659AcDF13311Ebd6183',
            'coin':'USDT',
            'metadata':{
                'type':'Crypto transfer'
            },
            'blockchain':'Binance Smart Chain'
        }
transferCrypto = lazerpay.transferCrypto(p_4)
print("transferCrypto =>")
print(transferCrypto)
p_5 = {
            'reference':'just a string',
            'amount':10.00,
            'fromCoin':'USDC',
            'toCoin':'USDT',
            'metadata':{
                'type':'Crypto Swap'
            },
            'blockchain':'Binance Smart Chain'
        }
swapCrypto = lazerpay.swapCrypto(p_5)
print("swapCrypto = >")
print(swapCrypto)
p_6 = {
            'reference':'just a string',
            'amount':10.00,
            'fromCoin':'USDC',
            'toCoin':'USDT',
            'metadata':{
                'type':'Crypto Swap'
            },
            'blockchain':'Binance Smart Chain'
        }
swapCharge = lazerpay.swapCharge(p_6)
print("swapCharge =>")
print(swapCharge)
getRate = lazerpay.getRate('USDT','NGN')
print("getRate =>")
print(getRate)
createRandomAccount = lazerpay.createRandomAccount()
print('createRandomAccount=>')
print(createRandomAccount)
getBalance = lazerpay.getBalance("0xD275Da57B35089a51A3E4659AcDF13311Ebd6183","to")#toWei. Use from to convert from wei to ether
print("getBalance =>")
print(getBalance)
print("__DONE__")
